guardrails/run/async_stream_runner.py

```diff
@@ -0,0 +1,567 @@
+            api=self.api,
+            instructions=instructions,
+            prompt=prompt,
+            msg_history=msg_history,
+            prompt_params=prompt_params,
+            prompt_schema=prompt_schema,
+            instructions_schema=instructions_schema,
+            msg_history_schema=msg_history_schema,
+            output_schema=output_schema,
+            output=self.output,
+            call_log=call_log,
+        )
+        async for call in result:
+            # yield ValidationOutcome[OT].from_guard_history(call)
+            yield call
+
+    # @async_trace(name="step")
+    async def async_step(
+        self,
+        index: int,
+        api: Optional[AsyncPromptCallableBase],
+        instructions: Optional[Instructions],
+        prompt: Optional[Prompt],
+        msg_history: Optional[List[Dict]],
+        prompt_params: Dict,
+        prompt_schema: Optional[StringSchema],
+        instructions_schema: Optional[StringSchema],
+        msg_history_schema: Optional[StringSchema],
+        output_schema: Schema,
+        call_log: Call,
+        output: Optional[str] = None,
+    ) -> AsyncGenerator[ValidationOutcome[OT], None]:
+        inputs = Inputs(
+            llm_api=api,
+            llm_output=output,
+            instructions=instructions,
+            prompt=prompt,
+            msg_history=msg_history,
+            prompt_params=prompt_params,
+            num_reasks=self.num_reasks,
+            metadata=self.metadata,
+            full_schema_reask=self.full_schema_reask,
+        )
+        outputs = Outputs()
+        iteration = Iteration(inputs=inputs, outputs=outputs)
+        call_log.iterations.push(iteration)
+        if output:
+            instructions = None
+            prompt = None
+            msg_history = None
+        else:
+            instructions, prompt, msg_history = await self.async_prepare(
+                call_log,
+                index,
+                instructions,
+                prompt,
+                msg_history,
+                prompt_params,
+                api,
+                prompt_schema,
+                instructions_schema,
+                msg_history_schema,
+                output_schema,
+            )
+
+        iteration.inputs.prompt = prompt
+        iteration.inputs.instructions = instructions
+        iteration.inputs.msg_history = msg_history
+
+        llm_response = await self.async_call(
+            index, instructions, prompt, msg_history, api, output
+        )
+        try:
+            stream_output = llm_response.async_stream_output
+        except AttributeError:
+            try:
+                stream_output = llm_response.stream_output
+
+            except AttributeError:
+                stream_output = llm_response.stream_output
+        if stream_output is None:
+            raise ValueError(
+                "No stream was returned from the API. Please check that "
+                "the API is returning an async generator."
```

Comment:
This is why we previously didn't support streaming for `Guard.parse`; i.e. nothing to stream.